plasma_core/child_chain.py

```diff
@@ -0,0 +1,119 @@
+from plasma_core.utils.transactions import decode_utxo_id, encode_utxo_id
+from plasma_core.utils.address import address_to_hex
+from plasma_core.constants import NULL_SIGNATURE
+from plasma_core.exceptions import (InvalidBlockSignatureException,
+                                    InvalidTxSignatureException,
+                                    TxAlreadySpentException,
+                                    TxAmountMismatchException)
+
+
+class ChildChain(object):
+
+    def __init__(self, operator):
+        self.operator = operator
+        self.blocks = {}
+        self.parent_queue = {}
+        self.child_block_interval = 1000
+        self.next_child_block = self.child_block_interval
+        self.next_deposit_block = 1
+
+    def add_block(self, block):
+        # Is the block being added to the head?
+        is_next_child_block = block.number == self.next_child_block
+        if is_next_child_block or block.number == self.next_deposit_block:
+            # Validate the block.
+            try:
+                self.__apply_block(block)
```

Comment:
calling `apply_block` actually would have side effect on changing the statement of block/tx. If it changes the input state here but return False might/might not lead to error on who calls this function, and would be super hard to understand what changes the state.
Probably verification and apply should be totally separate functions, and only the `True` case should call apply function. How do you think?